nntrainer/tensor/manager.h

```diff
@@ -108,6 +111,22 @@ class MMapedMemory {
 class Manager {
 
 public:
+  /**
+   * @brief Tensor Group Type
+   * @note this is not mutually exclusive list, a tensor might be identified as
+   * input as well as output
```

Comment:
in order to identify as both input and output, do we keep a list of this enum?
IMO INPUT and OUTPUT can be combined to one as a tensor allocated by tensor_pool and is an INPUT will always be OUTPUT to some other layer. So, all INPUT tensors will be OUTPUT for some layers (ignoring the special cases of the last layer where OUTPUT will not be INPUT).

Reply:
If the purpose is to just identify the request, then consider renaming to TensorRequestUsageType?

Reply:
> in order to identify as both input and output, do we keep a list of this enum?
IMO INPUT and OUTPUT can be combined to one as a tensor allocated by tensor_pool and is an INPUT will always be OUTPUT to some other layer. So, all INPUT tensors will be OUTPUT for some layers (ignoring the special cases of the last layer where OUTPUT will not be INPUT).

I agree, we can have only input tensors, let's migrate to having only single concept for this.

TensorUsageType also sounds good.